### PR TITLE
Add batch dim idx to support latest deepspeed DistributedAttention 

### PR DESCRIPTION
**Pull request by bhargaveede**
https://github.com/microsoft/DeepSpeed/commits/master/deepspeed/sequence/layer.py
With latest changes in DistributedAttention, we need to add batch_dim_idx and rotary_pos_emb to Distributed Attention.

This change brings changes to add the batch_dim_idx in line with new changes.

https://github.com/microsoft/DeepSpeed/commit/ffe0af23575c4f03a07408eacfc50b1a58781429

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by bhargaveede** [comment]
@regisss 
Distributed Attention implementation has been changed, and function signature is changed in Synapse 1.20.
To use Distributed Attention, we have to provide batch dimension and additional rope parameter from the model side.

Please review it and merge along with other 1.20 Dependent PRs

File: optimum/habana/transformers/models/llama/modeling_llama.py [hunk 1]
```diff
@@ -427,7 +427,64 @@ def forward(self, cur, dim, idx):
         return self.update(self.cache, cur, dim, idx, self.inp_seq_len)
 
 
-def GaudiDistributedAttention(fused_scaled_dot_product_attention, fused_scaled_dot_product_attention_distributed):
+class GaudiDistributedAttention(torch.nn.Module):
+    def __init__(self, hpu_module_fsdpa, scale, attention_dropout, enable_recompute, flash_attention_fp8):
```
Comment (mlapinskix) [inline optimum/habana/transformers/models/llama/modeling_llama.py:431]: Consider adding typehints.

File: optimum/habana/transformers/models/llama/modeling_llama.py [hunk 2]
```diff
@@ -427,7 +427,64 @@ def forward(self, cur, dim, idx):
...
+        super().__init__()
+        self._hpu_module_fsdpa = hpu_module_fsdpa
+        if parallel_state.sequence_parallel_is_initialized() and parallel_state.get_sequence_parallel_world_size() > 1:
+            from deepspeed.sequence.layer import DistributedAttention
+
+            self._hpu_module_fsdpa_distributed = DistributedAttention(
+                self._hpu_module_fsdpa, parallel_state.get_sequence_parallel_group(), 1, 2
+            )
+
+    def forward(
```
Comment (mlapinskix) [inline optimum/habana/transformers/models/llama/modeling_llama.py:441]: As above.

File: optimum/habana/transformers/models/llama/modeling_llama.py [hunk 3]
```diff
@@ -427,7 +427,64 @@ def forward(self, cur, dim, idx):
...
+                is_casual,
+                scale,
+                softmax_mode,
+                recompute_mode,
+                valid_sequence_lengths,
+                padding_side,
+            )
+
+
+def GetGaudiDistributedAttention(fused_scaled_dot_product_attention, fused_scaled_dot_product_attention_distributed):
```
Comment (regisss) [inline optimum/habana/transformers/models/llama/modeling_llama.py:487]:
```suggestion
def get_gaudi_distributed_attention(fused_scaled_dot_product_attention, fused_scaled_dot_product_attention_distributed):
```
I missed that last time, but this is a method so the letter case must be different.

**Review by regisss — COMMENTED** [review]
I left one comments to address + @mlapinskix's comments

**Comment by bhargaveede** [comment]
@regisss Addressed review comments. Can you review and merge

**Review by regisss — APPROVED** [review]
LGTM!
